tasks/github_tasks.py

```diff
@@ -16,11 +18,23 @@
 from tasks.release import _get_release_json_value
 
 
+@lru_cache(maxsize=None)
+def concurrency_key():
+    branch_name = get_git_branch_name()
```

Comment:
I just noticed a little quirk with this method. [get_git_branch_name](https://github.com/DataDog/datadog-agent/blob/00b27d48c0dfb3cf41d05ee30c9fd43b647b88b4/tasks/utils.py#L295-L299) does `git rev-parse --abbrev-ref HEAD`, which behaves like this:

```
$ git switch main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
$ git rev-parse --abbrev-ref HEAD
main

$ git switch 7.51.x
Switched to branch '7.51.x'
Your branch is up to date with 'origin/7.51.x'.
$ git rev-parse --abbrev-ref HEAD
7.51.x

$ git switch --detach 7.51.0
HEAD is now at 5b3c5ccb39 Finalize 7.51.0 (#22852)
$ git rev-parse --abbrev-ref HEAD
HEAD
```

Therefore, what might happen is that all builds on tagged commits end up using the same `HEAD` concurrency key.

In practice, I think it doesn't matter that much because it's quite rare to have concurrent tagged builds (RCs, final builds, custom builds), but if we want/need to prevent that, this method should check whether we are on a tag or not.

Reply:
Done!